### PR TITLE
Support Python 2 in SyncBackend

### PR DESCRIPTION
**Pull request by pquentin**
The changes:

* Hack around lack of nonlocal by using a context dictionary
* Fix a syntax error in Python 2 by moving **kwargs at the end of the parameters in a function call
* To make flake8 happy and avoid a cyclic dependency, I moved `LoopAbort` to `exceptions._LoopAbort` and only fill `backends.__init__.__all__` with `SyncBackend` in Python 2.

To test, use https://gist.github.com/pquentin/188f139be4c1bd5731b54bb2f7c29e41 to convert urllib3 to sync. I use this in the urllib3 repository: "for f in \`find urllib3 -name '*.py'\`; do echo $f; python3 ./asyncbleach.py $f | sponge $f; done". We'll need a better story for this. A better `asyncbleach.py` would at least know how to look at files recursively, but testing changes would still be annoying.

Then, run `python2 python2_demo.py`. I separated the changes in various commits, I'm not sure the nonlocal change is acceptable, maybe we should switch from closures to a class.

### CODE REVIEW

File: urllib3/backends/__init__.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
-class LoopAbort(Exception):
-    pass
-
-from .trio_backend import TrioBackend
+from urllib3.packages import six
 from .sync_backend import SyncBackend
-from .twisted_backend import TwistedBackend
+
+__all__ = [SyncBackend]
```
Comment (njsmith) [inline urllib3/backends/__init__.py]: `__all__` is supposed to be a list of strings, not objects :-)

File: urllib3/backends/__init__.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+
+if six.PY3:
```
Comment (njsmith) [inline urllib3/backends/__init__.py]: We need a better way to figure out if we're bleached or not... among other things, we may want to run the bleached code on python 3 for testing and other reasons. Maybe we can write `_BLEACHED = False`, and then have the bleaching script detect that specific set of tokens and rewrite it to `_BLEACHED = True`? Anyway, that's just a note for later...

File: urllib3/backends/sync_backend.py [hunk 3]
```diff
@@ -2,11 +2,11 @@
 import select
 import socket
 import ssl
+from ..exceptions import _LoopAbort
```
Comment (njsmith) [inline urllib3/backends/sync_backend.py]: I think `urllib3.exceptions` is supposed to be a piece of public API... maybe it'd be better to move `LoopAbort` into `urllib3.backends._util`, which has random bits of code that's common across the backends? (And then we could import it in `urllib3/backends/__init__.py`. Maybe `_util` should be renamed `_common` or something.)

File: urllib3/backends/__init__.py (hunk 1 above)
Comment (pquentin) [reply urllib3/backends/__init__.py]: Oops!

File: urllib3/backends/__init__.py (hunk 2 above)
Comment (pquentin) [reply urllib3/backends/__init__.py]: Agreed, that would be much better. Leaving that for a future pull request if you do not mind.

**Comment by pquentin** [comment]
Thanks for the insightful review. I updated the pull request, please take a look.

**Comment by njsmith** [comment]
Thanks!

**Pull request merged**